src/main/java/org/jboss/remoting3/remote/ClientConnectionOpenListener.java

```diff
@@ -401,7 +401,7 @@ public void handleEvent(final ConduitStreamSourceChannel channel) {
                         factoryOperator = and(ClientConnectionOpenListener.this.saslClientFactoryOperator, factoryOperator);
                         final SaslClient saslClient;
                         try {
-                            saslClient = configurationClient.createSaslClient(uri, ClientConnectionOpenListener.this.configuration, serverSaslMechs, factoryOperator);
+                            saslClient = createSaslClient(configurationClient, uri, ClientConnectionOpenListener.this.configuration, serverSaslMechs, factoryOperator);
```

Comment:
That's not why the failure happens; the failure happens because (for some reason) Elytron can't read its own content.

I'm afraid this issue is going to need some detailed analysis: in particular we need to provide a fix so that Elytron can read itself but at the same time we need to avoid assigning any superfluous permissions or leaking the execution context while doing so.